src/main/java/uk/gov/pay/ledger/report/resource/PerformanceReportResource.java

```diff
@@ -32,21 +34,36 @@ public PerformanceReportResource(PerformanceReportDao performanceReportDao) {
     @GET
     @Timed
     public PerformanceReportEntity getPerformanceReport(@QueryParam("from_date") String fromDate,
-                                                        @QueryParam("to_date") String toDate) {
+                                                        @QueryParam("to_date") String toDate,
+                                                        @QueryParam("state") String state) {
 
+        var params = addDateRangeParamsOrThrow(PerformanceReportParamsBuilder.builder(), fromDate, toDate);
+        params = addStateParamOrThrow(params, state);
+        return performanceReportDao.performanceReportForPaymentTransactions(params.build());
+    }
+
+    private PerformanceReportParamsBuilder addStateParamOrThrow(PerformanceReportParamsBuilder builder, String state) {
+        if (isNotBlank(state)) {
+            Optional.ofNullable(TransactionState.from(state)).ifPresentOrElse(
```

Comment:
Very minor but I think there is an `orElseThrow` on the `Optional` construct, that might ready nicely. 

Something like: 

```java 
Optional.ofNullable(TransactionState.from(state))
  .map(builder::withState)
  .orElseThrow(() -> new ValidationException("..."))
```

Reply:
This suggestion is better; I've updated the code